### PR TITLE
chore: vitest

### PR DESCRIPTION
**Pull request by alvarosabu**
Migrating from jest to vitest

### CODE REVIEW

File: lib/package.json [hunk 1]
```diff
@@ -31,17 +34,19 @@
     "@cypress/vite-dev-server": "^5.0.5",
     "@cypress/vue": "^5.0.5",
     "@vitejs/plugin-vue": "^4.2.1",
+    "@vitest/ui": "^0.31.0",
     "@vue/babel-preset-app": "^5.0.8",
-    "@vue/test-utils": "2.3.2",
+    "@vue/test-utils": "^2.3.2",
     "@vue/tsconfig": "^0.1.3",
     "@vue/vue3-jest": "^29.2.3",
     "babel-jest": "^29.4.3",
```
Comment (alexjoverm) [inline lib/package.json:42]: I think we don't need this anymore

File: lib/package.json [hunk 2]
```diff
@@ -31,17 +34,19 @@
     "@cypress/vite-dev-server": "^5.0.5",
     "@cypress/vue": "^5.0.5",
     "@vitejs/plugin-vue": "^4.2.1",
+    "@vitest/ui": "^0.31.0",
     "@vue/babel-preset-app": "^5.0.8",
```
Comment (alexjoverm) [inline lib/package.json:38]: I think we don't need this dep, neither `@babel/core` or any Babel related as it was only used by Jest

File: lib/test/mocks/storyblok-client.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class MockStoryblokClient {
```
Comment (alexjoverm) [inline lib/test/mocks/storyblok-client.ts:1]: See coment above

File: package.json [hunk 4]
```diff
@@ -25,6 +25,7 @@
     "@typescript-eslint/parser": "^5.59.2",
     "eslint": "^8.38.0",
     "eslint-config-prettier": "^8.8.0",
+    "eslint-plugin-vitest-globals": "^1.3.1",
```
Comment (alexjoverm) [inline package.json:28]: see coment above

**Review by alexjoverm — CHANGES_REQUESTED** [review]
@alvarosabu amazing work here. I left a few small comments + couple of suggestions. Let me know what you think.

Btw I love that vitest/ui ;)

File: lib/package.json (hunk 1 above)
Comment (alvarosabu) [reply lib/package.json:42]: Depends if we want to mount components on the unit test or not, if we agree to leave that to the e2e then yes we can remove. @alexjoverm Let me know

File: lib/test/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    app.use(StoryblokVue, {});
+
+    // Check if the plugin has been installed properly
+    expect(app.component("StoryblokComponent")).toBeTruthy();
+    expect(app._context.directives.editable).toBeTruthy();
+  });
+
+  /* test("useStoryblok", async () => {
+
```
[the remark this answers is not in the document]
Comment (alvarosabu) [reply lib/test/index.test.ts]: @alexjoverm That being said then, does it make sense to have Vitest installed at all?

I had the idea that we were going to increase the coverage unit testing by taking advantage of migrating to Vitest but if we prefer to focus on e2e then let's remove it.

Let me know

**Pull request closed without merging**